### PR TITLE
Implement workout section list

### PR DESCRIPTION
**Pull request by jpoist97**
Implemented section list for workout component and EllipsisPopup

We also need to decide how we want to do the colors now that this list is sorted and split into sections

### CODE REVIEW

**Comment by jpoist97** [comment]
IMG_3767

**Comment by jpoist97** [comment]
Screenshot of popup
IMG_3768

File: components/workouts/AllWorkouts.jsx [hunk 1]
```diff
@@ -11,29 +13,100 @@ const Title = styled.Text`
...
+    {right ? (
+      <WorkoutCard
+        name={right.name}
+        subtext={right.subtext}
+        displayEllipses={right.displayEllipses}
+        deleteWorkout={right.deleteWorkout}
+        onPress={right.onPress}
+        color={right.color}
+        key={right.name + right.subtext}
+      />
```
Comment (ctperry0301) [inline components/workouts/AllWorkouts.jsx:82]: I don't have a problem with the "left cards & right cards" way. It's pretty easy to understand and it works as expected. 
That being said, why did you do it like this as opposed to just using styling and numColumns?

**Review by ctperry0301 — DISMISSED** [review]
Looks good.

File: components/workouts/AllWorkouts.jsx (hunk 1 above)
Comment (jpoist97) [reply components/workouts/AllWorkouts.jsx:82]: Unless I missed something in the documentation, there isn't numColumns for the alphabet list component I used for this. It also wasn't there for any of the other options I looked at, although documentation wasn't always great.

**Review by ctperry0301 — APPROVED** [review]

File: components/workouts/AllWorkouts.jsx [hunk 2]
```diff
@@ -11,29 +13,100 @@ const Title = styled.Text`
...
+    } else {
+      accumulator[bucket].push(item);
+    }
+
+    return accumulator;
+  }, {});
+
+  // Pair up each bucket of data
+  const pairedBucketData = _.mapValues(bucketData, (data) => {
+    const pairData = [];
```
Comment (shriyan44) [inline components/workouts/AllWorkouts.jsx:50]: What does pairing up mean here?

**Review by shriyan44 — APPROVED** [review]
good job troni

**Pull request merged**